src/components/PageBlocks/GeometryBase.vue

```diff
@@ -0,0 +1,186 @@
+<template>
+  <wrap
+    v-bind="$props"
+    v-on="$listeners"
+  >
+    <div
+      v-if="processing"
+      class="d-flex align-items-center justify-content-center h-100"
+    >
+      <b-spinner />
+    </div>
+    <template v-else>
+      <l-map
+        ref="map"
+        :zoom="map.zoom"
+        :center="map.center"
+        :min-zoom="map.zoomMin"
+        :max-zoom="map.zoomMax"
+        :bounds="map.bounds"
+        class="w-100"
+        style="height: 100%;"
+        :max-bounds="map.lockBounds ? map.bounds : null"
+      >
+        <l-tile-layer
+          url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
+          :attribution="map.attribution"
+        />
+        <l-polygon
+          v-for="(geometry, i) in geometries"
+          :key="`polygon-${i}`"
+          :lat-lngs="geometry.map(value => value.geometry)"
+          :color="colors[i]"
+        />
+
+        <l-marker
+          v-for="(marker, i) in localValue"
+          :key="`marker-${i}`"
+          :lat-lng="marker.value"
+          :icon="getIcon(marker)"
+        />
+      </l-map>
+    </template>
+  </wrap>
+</template>
+
+<script>
+import { divIcon, latLng, latLngBounds } from 'leaflet'
+import {
+  LPolygon,
+} from 'vue2-leaflet'
+import { compose, NoID } from '@cortezaproject/corteza-js'
+import { mapGetters, mapActions } from 'vuex'
+import { evaluatePrefilter } from 'corteza-webapp-compose/src/lib/record-filter'
+import base from './base'
+
+export default {
+  components: { LPolygon },
+
+  extends: base,
+
+  data () {
+    return {
+      map: {
+        zoom: 4,
+        center: [47.31322, -1.319482],
+      },
+
+      processing: false,
+      show: false,
+
+      geometries: [],
+      colors: [],
+      markers: [],
+    }
+  },
+
+  computed: {
+    ...mapGetters({
+      getModuleByID: 'module/getByID',
+    }),
+
+    localValue () {
+      const values = []
+
+      this.geometries.forEach((geo) => {
+        geo.forEach((value) => {
+          values.push({ value: this.getLatLng(value.geometry), color: value.color })
+        })
+      })
+
+      return values
+    },
+
+    bounds () {
+      let { boundLowerRight, boundTopLeft } = this.options
+
+      boundTopLeft = this.parseGeometryField(boundTopLeft)
+      boundLowerRight = this.parseGeometryField(boundLowerRight)
+
+      return latLngBounds([this.getLatLng(boundTopLeft), this.getLatLng(boundLowerRight)])
+    },
+  },
+
+  created () {
+    this.loadEvents()
```

Comment:
Instead of on created, have two watchers, one for 'record.recordID', the other for options. Look at RecordListBase line 840. This will ensure the map is re-rendered when options change or we come to a new page ('record.recordID' watcher)